src/scatter.jl

```diff
@@ -0,0 +1,44 @@
+ATM_OPS = Dict((+) => CUDA.atomic_add!, (-) => CUDA.atomic_sub!, (max) => CUDA.atomic_max!, (min) => CUDA.atomic_min!,
+               (*) => CUDA.atomic_mul!, (/) => CUDA.atomic_div!, (&) => CUDA.atomic_and!, (|) => CUDA.atomic_or!)
+
+for (op, atm_op) in ATM_OPS
+    @eval function scatter_kernel!(op::typeof($(op)), dst, src, idx)
+        index = threadIdx().x + (blockIdx().x - 1) * blockDim().x
+
+        @inbounds if index <= length(idx)
+            i = Base._to_linear_index(dst, idx[index]...)
+            $(atm_op)(pointer(dst, i), src[index])
```

Comment:
if you use `@atomic` you don't need to call `pointer` or calculate a linear index yourself

Reply:
Done.